packages/contentstack-import/src/import/modules/extensions.ts

```diff
@@ -0,0 +1,124 @@
+import isEmpty from 'lodash/isEmpty';
+import values from 'lodash/values';
+import {join, resolve } from 'node:path';
+import { FsUtility } from '@contentstack/cli-utilities';
+
+import config from '../../config';
+import { log, formatError } from '../../utils';
+import BaseClass, { ApiOptions } from './base-class';
+import { ModuleClassParams, Extensions } from '../../types';
+
+export default class ImportExtensions extends BaseClass {
+  private fs: FsUtility;
+  private mapperDirPath: string;
+  private extensionsFolderPath: string;
+  private extUidMapperPath: string;
+  private extSuccessPath: string;
+  private extFailsPath: string;
+  private extensionsConfig: Extensions;
+  private extensions: Record<string, unknown>;
+  private extUidMapper: Record<string, unknown>;
+  private extSuccess: Record<string, unknown>[];
+  private extFailed: Record<string, unknown>[];
+
+  constructor({ importConfig, stackAPIClient }: ModuleClassParams) {
+    super({ importConfig, stackAPIClient });
+
+    this.mapperDirPath = join(this.importConfig.backupDir, 'mapper', 'extensions');
+    this.extensionsFolderPath = join(this.importConfig.backupDir, this.extensionsConfig.dirName);
+    this.extUidMapperPath = join(this.mapperDirPath, 'uid-mapping.json');
+    this.extSuccessPath = join(this.importConfig.backupDir, 'extensions', 'success.json');
+    this.extFailsPath = join(this.importConfig.backupDir, 'extensions', 'fails.json');
+    this.fs = new FsUtility({ basePath: this.mapperDirPath });
+    this.extensions = this.fs.readFile(
+      join(this.importConfig.backupDir, 'extensions', 'extensions.json'),
+      true,
+    ) as Record<string, unknown>;
+    this.extFailed = [];
+    this.extSuccess = [];
+    this.extUidMapper = {};
+    this.extensionsConfig = config.modules.extensions;
+  }
+
+  /**
+   * @method start
+   * @returns {Promise<void>} Promise<void>
+   */
+  async start(): Promise<void> {
+    log(this.importConfig, 'Migrating extensions', 'info');
+
+    this.extUidMapper = this.fs.readFile(join(this.extUidMapperPath), true) as Record<string, unknown> || {};
+
+    await this.importExtensions();
+
+    if (this.extSuccess?.length) {
+      new FsUtility({ basePath: this.extensionsFolderPath }).writeFile(this.extSuccessPath, this.extSuccess);
+    }
+
+    if (this.extFailed?.length) {
+      new FsUtility({ basePath: this.extensionsFolderPath }).writeFile(this.extFailsPath, this.extFailed);
+    }
+  }
+
+  async importExtensions(): Promise<any> {
+    if (this.extensions === undefined || isEmpty(this.extensions)) {
+      log(this.importConfig, 'No Extensions Found', 'info');
+      return resolve();
+    }
+
+    const apiContent = values(this.extensions);
+
+    const onSuccess = ({ response, apiData: { uid, title } = { uid: null, title: '' } }: any) => {
+      this.extSuccess.push(response);
```

Comment:
Is this code working, can you check this ?

Reply:
this.extSuccess.push(response); this.extUidMapper[uid] = response.uid;,  this context would be from parent base class and it doesnt have these properties. did it work ?